pypeit/core/telluric.py

```diff
@@ -135,11 +134,69 @@ def pca_eval(theta,pca_dict):
 #        theta_pca[2:npca+1]``
 #
 #    """
-#    gaussian_mixture_model = pca_dict['prior']
+#    gaussian_mixture_model = qso_pca_dict['prior']
 #    A = theta[2:]
 #    return gaussian_mixture_model.score_samples(A.reshape(1,-1))
 
 
+def read_telluric_pca(filename, wave_min=None, wave_max=None, pad_frac=0.10):
+    """
+    Reads in the telluric PCA components from a file.
+
+    Optionally, this method also trims wavelength to be in within ``wave_min``
+    and ``wave_max`` and pads the data (see ``pad_frac``).
+
+    .. todo::
+        List and describe the contents of the dictionary in the return
+        description.
+
+    Args:
+        filename (:obj:`str`):
+            Telluric PCA filename
+        wave_min (:obj:`float`):
+            Minimum wavelength at which the grid is desired
+        wave_max (:obj:`float`):
+            Maximum wavelength at which the grid is desired.
+        pad_frac (:obj:`float`):
+            Percentage padding to be added to the grid boundaries if
+            ``wave_min`` or ``wave_max`` are input; ignored otherwise. The
+            resulting grid will extend from ``(1.0 - pad_frac)*wave_min`` to
+            ``(1.0 + pad_frac)*wave_max``.
+
+    Returns:
+        :obj:`dict`: Dictionary containing the telluric PCA components.
+            - wave_grid=
+            - dloglam=
+            - resln_guess=
+            - pix_per_sigma=
+            - tell_pad_pix=
+            - ncomp_tell_pca=
+            - tell_pca=
+            - bounds_tell_pca=
+            - coefs_tell_pca=
+    """
+    # load_telluric_grid() takes care of path and existance check
+    hdul = data.load_telluric_grid(filename)
+    wave_grid_full = hdul[1].data
+    pca_comp_full = hdul[0].data
+    ncomp = hdul[0].header['NCOMP']
+    bounds = hdul[2].data
+    model_coefs = hdul[3].data
+
+    ind_lower = np.argmin(np.abs(wave_grid_full - (1.0 - pad_frac)*wave_min)) \
+                    if wave_min is not None else 0
+    ind_upper = np.argmin(np.abs(wave_grid_full - (1.0 + pad_frac)*wave_max)) \
+                    if wave_max is not None else nspec_full
+    wave_grid = wave_grid_full[ind_lower:ind_upper]
+    pca_comp_grid = pca_comp_full[:,ind_lower:ind_upper]
+
+    dwave, dloglam, resln_guess, pix_per_sigma = wvutils.get_sampling(wave_grid)
+    tell_pad_pix = int(np.ceil(10.0 * pix_per_sigma))
+
+    return dict(wave_grid=wave_grid, dloglam=dloglam, resln_guess=resln_guess,
+            pix_per_sigma=pix_per_sigma, tell_pad_pix=tell_pad_pix, ncomp_tell_pca=ncomp,
+            tell_pca=pca_comp_grid, bounds_tell_pca=bounds, coefs_tell_pca=model_coefs)
+            
```

Comment:
Can you make it clear that the read_telluric grid is now deprecated compared to the pca approach in the docs to read_telluric_grid